web/modules/custom/hoeringsportal_audit_log/src/Form/SettingsForm.php

```diff
@@ -0,0 +1,160 @@
+   */
+  protected function getEditableConfigNames(): array {
+    return [
+      self::SETTINGS,
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @phpstan-param array<mixed, mixed> $form
+   * @phpstan-return array<mixed, mixed>
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    // Form constructor.
+    $form = parent::buildForm($form, $form_state);
+    // Default settings.
+    $config = $this->config(self::SETTINGS);
+
+    $url = Url::fromRoute('os2web_audit.plugin_settings_local_tasks');
+    $form['info'] = [
+      '#markup' => '<p>These configurations handle <i>when</i> to create logs, <a href="' . $url->toString() . '">the <code>os2web_audit</code> configuration can be found here.</a></p>',
+      '#title' => $this->t('Logged pages'),
+    ];
+
+    $form['logged_pages'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Logged pages'),
+      '#tree' => TRUE,
+    ];
+
+    $form['logged_pages']['logged_route_names'] = [
+      // @todo get all routes (a lot), find a way to filter them and display them in a multiselect.
+      '#type' => 'textarea',
+      '#title' => $this->t('Route names'),
+      '#default_value' => $config->get('logged_route_names') ? $this->fromArrayToString($config->get('logged_route_names')) : NULL,
+      '#description' => $this->t('Comma seperated list. Route names to log when users visit, they can look like this: <code>hoeringsportal_citizen_proposal.admin_supporter</code>, if in doubt, ask your friendly neighborhood programmer.'),
+    ];
+
+    $form['logged_content_types'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Logged content types'),
+      '#tree' => TRUE,
+    ];
+
+    // Add config for all configured node types.
+    $nodeTypes = NodeType::loadMultiple();
+
+    foreach ($nodeTypes as $nodeType) {
+      $form['logged_content_types'][$nodeType->id()] = [
+        '#type' => 'fieldset',
+        '#title' => $nodeType->label(),
+        '#tree' => TRUE,
+      ];
+
+      $defaultValues = $config->get('logged_content_types');
+
+      // Make it possible to log when a user is on the view page of
+      // this content type.
+      $form['logged_content_types'][$nodeType->id()]['view'] = [
+        '#type' => 'checkbox',
+        '#title' => $this->t('Log view'),
+        '#description' => $this->t('Log when a user views this content page (<code>/node/{node_id}</code>'),
+        '#default_value' => $defaultValues[$nodeType->id()]['view'] ?? NULL,
+      ];
+
+      // Make it possible to log when a user is on the edit page of
+      // this content type.
+      $form['logged_content_types'][$nodeType->id()]['edit'] = [
+        '#type' => 'checkbox',
+        '#title' => $this->t('Log edit'),
+        '#description' => $this->t('Log when a user views the edit content page (<code>/node/{node_id}/edit</code>'),
+        '#default_value' => $defaultValues[$nodeType->id()]['edit'] ?? NULL,
+      ];
+    }
```

Comment:
What about “delete”? Should we log when a node is deleted?

Reply:
hmm.. We could. we also could for create. but I am not sure it makes sense.